File: Pranai/Strings/Valid_Palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        import re
        s1 = re.sub('[^a-zA-Z0-9]',' ',s)
        s1 = s1.lower()
        s1 = s1.split()
        s1 = ''.join(s1)
        s2 = s1[::-1]
        #print(s1,s2)
        if(s1==s2):
            return True
        else:
            return False

File: Krishna/Strings/Valid Palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s)
    {
        int l=s.length();
        for(int i=0;i<l;i++)
        s[i]=tolower(s[i]);
        int j=0;
        while(j<l){
            if(!(iswalnum(s[j])))              
                j++;
            else if(!(iswalnum(s[l])))
                l--;
            else if(s[j]==s[l])
                j++,l--;
            else
                return false;
            }        
        return true;
        
    }
};
